From 6363d6cf9ad4c3fdd8ea4ecdcd998d7dd561e896 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Sat, 3 Jun 2017 20:10:46 -0700
Subject: [PATCH 1/2] Allow adb root even in no debuggable builds

this makes adb root available on user builds or
with Magisk installed (e.g. when using adb over wifi)

Change-Id: I507c6d90f0e355d8a931692f617b58b5f9eb6f20
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
Signed-off-by: Kazakov Stepan <ksrt12group@gmail.com>
---
 adb/daemon/main.cpp | 4 ++--
 adb/services.cpp    | 6 ------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/adb/daemon/main.cpp b/adb/daemon/main.cpp
index 729d5fbb7..89d31429a 100644
--- a/adb/daemon/main.cpp
+++ b/adb/daemon/main.cpp
@@ -89,9 +89,9 @@ static bool should_drop_privileges() {
         drop = true;
     }
 
-    return drop;
+    return false;
 #else
-    return true; // "adb root" not allowed, always drop privileges.
+    return false;
 #endif // ALLOW_ADBD_ROOT
 }
 
diff --git a/adb/services.cpp b/adb/services.cpp
index 9b1a07dd6..0f7c29bdf 100644
--- a/adb/services.cpp
+++ b/adb/services.cpp
@@ -78,12 +78,6 @@ void restart_root_service(int fd, void *cookie) {
         WriteFdExactly(fd, "adbd is already running as root\n");
         adb_close(fd);
     } else {
-        if (!__android_log_is_debuggable()) {
-            WriteFdExactly(fd, "adbd cannot run as root in production builds\n");
-            adb_close(fd);
-            return;
-        }
-
         android::base::SetProperty("havoc.service.adb.root", "1");
         WriteFdExactly(fd, "restarting adbd as root\n");
         adb_close(fd);
-- 
2.20.1

